src/views/login/components/login.js

```diff
@@ -4,16 +4,38 @@ import './login.scss';
 export default class Login extends Component {
 
     static propTypes = {
+        usernameChanged: PropTypes.func.isRequired,
+        passwordChanged: PropTypes.func.isRequired,
+        login: PropTypes.func.isRequired,
+
+        usernameField: PropTypes.string.isRequired,
+        passwordField: PropTypes.string.isRequired,
+        history: PropTypes.object.isRequired,
     };
 
+    handleUsernameChanged = (e) => {
+        const { usernameChanged } = this.props;
+        usernameChanged(e.target.value);
+    }
+
+    handlePasswordChanged = (e) => {
+        const { passwordChanged } = this.props;
+        passwordChanged(e.target.value);
+    }
+
+    handleLoginClicked = () => {
+        const { login, usernameField, passwordField, history } = this.props;
+        login(usernameField, passwordField, history);
```

Comment:
This ultimately links to the `login` function in `user-action-creators.js` right?  Though it seems to mediated through the `login` in `propTypes`.   I'm assuming that the `propTypes` have to be set when the component is created -- how does that work, and is this where the `...` operator comes into play?